core/src/proxy.c

```diff
@@ -108,16 +114,23 @@ struct knot_proxy *knot_proxy_register(u8_t id, const char *name,
 	struct knot_proxy *proxy;
 
 	/* Out of index? */
-	if (id >= CONFIG_KNOT_THING_DATA_MAX)
-		return NULL;
+	if (id >= CONFIG_KNOT_THING_DATA_MAX) {
+		LOG_ERR("ID %d: id >= CONFIG_KNOT_THING_DATA_MAX (%d)",
+			id, CONFIG_KNOT_THING_DATA_MAX);
+		return register_fail(id);
```

Comment:
`return NULL;`
You are already logging the reason why the register fails there is no need to log where was the fail. I think that with zephyr logger you can enable the callback trace if you really need to know what function raise this logging.

Reply:
Zephyr logs the function that failed it. But yet I think it is not explicit enough